@ecomplus/storefront-components/src/js/TheProduct.js

```diff
@@ -500,6 +513,40 @@ export default {
       this.fetchProduct()
     }
     this.isFavorite = checkFavorite(this.body._id || this.productId, this.ecomPassport)
+
+    // Just for number length check
+    const formatTime = (number) => {
+      return number.toString().length === 1 ? `0${number}` : number
+    }
+
+    // Retrieve promo timer
+    const getRemainingPromoTime = (c, t) => {
+      const distance = t - c;
+      // Uncomment days if you wish to show the full promo time
+      // const days = Math.floor(distance / (1000 * 60 * 60 * 24));
+      const hours = Math.floor((distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+      const minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
+      const seconds = Math.floor((distance % (1000 * 60)) / 1000);
+      return `${formatTime(hours)}:${formatTime(minutes)}:${formatTime(seconds)}`
+    }
+
+    // Retrieve timestamps
+    const currentTime = new Date().toISOString()
+    const targetTime = new Date(`${currentTime.split('T')[0]}T23:59:59`).getTime()
```

Comment:
Não entendi a lógica do seu `targetTime` aqui, quer dizer, ele deveria ser baseado na data final da promoção, não? Você mockou aqui pro dia atual...

Reply:
Além disso, isso só deveria ser executado se `this.isOnSale` (na sua implementação atual), caso contrário a constante é inútil

Reply:
Resolvido